### PR TITLE
qt 6.0.4

### PR DESCRIPTION
**Pull request by paperchalice**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/qt.rb [hunk 1]
```diff
@@ -24,9 +23,9 @@ class Qt < Formula
   depends_on "cmake" => [:build, :test]
   depends_on "ninja" => :build
   depends_on "pkg-config" => :build
-  depends_on xcode: [:build, :test]
 
   depends_on "assimp"
+  depends_on "brotli"
```
Comment (carlocab) [inline Formula/qt.rb:28]: Do you need to change this?

https://github.com/Homebrew/homebrew-core/blob/539e9f5445d2bab4ce636a7f798bd94e5a8d9afe/Formula/qt.rb#L86

**Review by carlocab — COMMENTED** [review]
Failed on Mojave. Is that expected?

**Comment by cho-m** [comment]
Mojave node is failing on 10.15 SDK API usage, e.g. [`charactersByApplyingModifiers`](https://developer.apple.com/documentation/appkit/nsevent/3242717-charactersbyapplyingmodifiers?language=objc)

It will probably require a minimum of Xcode 11 with SDK 10.15.

**Comment by carlocab** [comment]
> It will probably require a minimum of Xcode with SDK 10.15.

Mojave should have Xcode 11.4, so that should work with an Xcode dependency. Could gated with `if MacOS.version <= :mojave`.

**Comment by Bo98** [comment]
> It will probably require a minimum of Xcode with SDK 10.15.

Would did this change in a bug fix release? Sounds like a bug or something going wrong rather than anything intentional.

**Comment by cho-m** [comment]
> Would did this change in a bug fix release? Sounds like a bug or something going wrong rather than anything intentional.

This PR does remove `depends_on xcode: [:build, :test]`, if that impacts anything.

**Comment by Bo98** [comment]
Yes, that would make it switch from the Xcode 10.15 SDK to the CLT 10.14 SDK.

It's also why you suddenly needed a `ENV.delete "CPATH"` in your test.

**Comment by Bo98** [comment]
Arguably, the code should be wrapped in a `QT_MACOS_PLATFORM_SDK_EQUAL_OR_ABOVE(10.15)`. Not entirely sure why that isn't done.

**Pull request closed without merging**